herbarium-sheet-plant-organ-detection/main.py

```diff
@@ -22,74 +26,89 @@ def start_kafka(predictor: DefaultPredictor) -> None:
+        image_uri = json_value['object']['digitalEntity']['ac:accessURI']
+        additional_info_annotations, width, height = run_object_detection(image_uri, predictor)
         logging.info('Publishing the result: %s', json_value)
-        annotation = map_to_annotation(json_value, additional_info_annotations)
-        send_updated_opends(annotation, producer)
+        event = map_to_annotation(json_value, additional_info_annotations, width, height)
+        event["jobId"] = json_value['jobId']
+        send_updated_opends(event, producer)
 
 
-def map_to_annotation(json_value, additional_info_annotations) -> dict:
-    values = []
+def map_to_annotation(json_value, additional_info_annotations, width, height) -> dict:
+    annotations = []
     for value in additional_info_annotations:
-        values.append(
-            {
-                'class': value['class'],
-                'score': value['score'],
-                'selector': {
-                    'type': 'FragmentSelector',
-                    'conformsTo': 'http://www.w3.org/TR/media-frags/',
-                    'value': "xywh=" + str(value['boundingBox'][0]) + ',' + str(value['boundingBox'][1]) + ',' + str(
-                        value['boundingBox'][2]) + ',' + str(value['boundingBox'][3])
+        body_values = [{
+            'class': value['class'],
+            'score': value['score'],
+        }]
+        selector = {
+            'oa:selector': {
+                'ods:type': 'FragmentSelector',
+                'dcterms:conformsTo': 'https://www.w3.org/TR/media-frags/',
+                'ac:hasRoi': {
+                    "ac:xFrac": value['boundingBox'][0] / width,
+                    "ac:yFrac": value['boundingBox'][1] / height,
+                    "ac:widthFrac": (value['boundingBox'][2] - value['boundingBox'][0])/width,
+                    "ac:heightFrac": (value['boundingBox'][3] - value['boundingBox'][1]) / height
                 }
             }
-        )
-    annotation = {
-        'type': 'Annotation',
-        'motivation': 'https://hdl.handle.net/pid-machine-enrichment',
-        'creator': 'https://hdl.handle.net/enrichment-service-pid',
-        'created': timestamp_now(),
-        'target': {
-            'id': json_value['id'],
-            'type': 'https://hdl.handle.net/21...'
-        },
-        'body': {
-            'source': json_value['data']['ac:accessURI'],
-            'purpose': 'describing',
-            'values': values
         }
+        annotation = {
+            'rdf:type': 'Annotation',
+            'oa:motivation': 'ods:adding',
+            'oa:target': {
+                'ods:id': json_value['object']['digitalEntity']['ods:id'],
+                'ods:type': 'https://hdl.handle.net/21.T11148/bbad8c4e101e8af01115', # handle for digitalMedia Type
+                'oa:selector': selector
+            },
+            'oa:body': {
+                'ods:type': 'TextualBody',
```

Comment:
Where does this type come from?

Reply:
Epic type registry. It's the main place we're putting identifiers for Types.

Reply:
Wouldn't it be better to just pick this from the object itself? If we ever update the type we wouldn't have to remember that we have to change it here as well.

Reply:
What property of the object would you choose then? TextualBody refers to the value I guess, which is an array of strings in most cases, but can be different?

Reply:
Sorry tom, I thought you were referencing a different "ods:type" in your original comment. 

In the annotation RFC, [there's an example for FragmentSelector](https://docs.google.com/document/d/1jfiVLzgSkSLPSaTfe9JfhBEz1KVABy1NDqyNI7OaFKw/edit?disco=AAABAoPOFws). The body ods:type there is "TextualBody", so I copied that. Not sure what else it could be, I am open to suggestions.